src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -206,9 +206,7 @@ public function belongsTo($related, $foreignKey = null, $ownerKey = null, $relat
         // If no foreign key was supplied, we can use a backtrace to guess the proper
         // foreign key name by using the name of the relationship function, which
         // when combined with an "_id" should conventionally match the columns.
-        if (is_null($foreignKey)) {
-            $foreignKey = Str::snake($relation).'_'.$instance->getKeyName();
-        }
+        $foreignKey = $foreignKey ?: Str::snake($relation).'_'.$instance->getKeyName();
```

Comment:
?: = non falsey, so not the same as a null check

Reply:
Thank you for reviewing.

I know that ?: behaves differently than is_null.
https://www.php.net/manual/en/types.comparisons.php

But, in this method, $foreignKey only become string or null.
https://3v4l.org/8WYJ7

Even if an empty string is passed, it cannot be determined as is_null, but if it is ?:, it can be determined as false, and the foregin key can be guessed and assigned.

I am a beginner of the PHP.
Im sorry if there are mistakes.